### PR TITLE
Move authn context to istio/authn for sharing between filters.

### PR DESCRIPTION
**Pull request by diemtvu**
Move authn context proto to istio in order to use them in other filters (e.g mixer filter to consume authn output for check and report)

```NONE
```

### CODE REVIEW

File: src/istio/authn/BUILD [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package(default_visibility = ["//visibility:public"])
+
+load(
+    "@envoy//bazel:envoy_build_system.bzl",
+    "envoy_proto_library",
+)
+
+envoy_proto_library(
+    name = "context_proto",
```
Comment (qiwzhang) [inline src/istio/authn/BUILD]: The package name in the proto should be changed to match the folder path

Comment (diemtvu) [reply src/istio/authn/BUILD]: done

**Pull request merged**